content/en/install.md

```diff
@@ -3,101 +3,206 @@ title: Installation
+Here is a step-by-step guide to setting up a project to try SciPy with [`uv`],
+a Python package manager.
 
-<a name="pip-install"></a>
+[`uv`]: https://docs.astral.sh/uv/
 
-# Installing with Pip
+1. Install `uv`, following [the instructions in the `uv` documentation][install-uv].
 
-You can install SciPy from PyPI with `pip`:
+[install-uv]: https://docs.astral.sh/uv/getting-started/installation/
 
-    python -m pip install scipy
+2.  Create a new project in a new subdirectory, by executing the following in a terminal:
 
-<a name="conda-install"></a>
+        uv init try-scipy
+        cd try-scipy
 
-# Installing via Conda
+    {{< admonition hint >}}
+The second command changes directory into the directory of your project.
+    {{< /admonition >}}
 
-You can install SciPy from the `defaults` or `conda-forge` channels with `conda`:
+3.  Install Python:
 
-    conda install scipy
+        uv python install
 
-<a name="package_manager"></a>
+    {{< admonition note >}}
+If you have already installed Python, this step is optional.
+    {{< /admonition >}}
 
-# Install system-wide via a package manager
+4.  Add SciPy to your project:
+
+        uv add scipy
+
+5.  Try out SciPy!
+
+        uv run python
+
+    This will launch a Python interpreter session, from which you can `import scipy`.
+
+See next steps in [the SciPy user guide][scipy-user-guide].
+
+[scipy-user-guide]: https://docs.scipy.org/doc/scipy/tutorial/
+
+### Installing with `pixi`
+
+If you work with non-Python packages, you may prefer to install SciPy as
+a [Conda] package, so that you can use the same workflow for packages which
+are not available on [PyPI](https://pypi.org/), the Python Package Index.
+Conda can manage packages in any language, so you can use it to install
+Python itself, compilers, and other languages.
+
+[Conda]: https://docs.conda.io/projects/conda/en/latest/index.html
+
+The steps to install SciPy from [conda-forge] using the package management
+tool [`pixi`] are very similar to the steps for `uv`:
+
+[conda-forge]: https://conda-forge.org/
+[`pixi`]: https://pixi.sh/latest/
+
+1. Install `pixi`, following [the instructions in the `pixi` documentation][install-pixi].
+
+[install-pixi]: https://pixi.sh/latest/
+
+2.  Create a new project in a new subdirectory:
+
+        pixi init try-scipy
+        cd try-scipy
+
+3.  Add SciPy to your project:
+
+        uv add scipy
+
+    {{< admonition note >}}
+This step also adds Python from conda-forge.
+    {{< /admonition >}}
+
+4.  Try out SciPy!
+
+        pixi run python
+
+See next steps in [the SciPy user guide][scipy-user-guide].
+
+<a name="environment-based"></a>
+
+## Environment-based workflows
+
+In environment-based workflows, you install packages into an environment, which you
+can activate and deactivate.
+These workflows are well-established,
+but lack some reproducibility benefits of project-based workflows.
+
+### Installing with `pip`
+
+1.  [Install Python](https://www.python.org/downloads/).
```

Comment:
Of course, you can use `uv` to do this as well and then switch back to being a simple `venv` dinosaur like me. But there are indeed already some comments about choice paralysis I think.

Reply:
Yeah, I considered leaving out the hyperlink and letting users google "how to install Python" - could do that if people prefer. I don't think we should be laying out different options for installing Python.

Reply:
It looks like the uv folks are urging people to [switch away from activating virtual environments](https://github.com/astral-sh/uv/issues/1910#issuecomment-2498847428). And they have good technical reasons for it.

That's all fine if you've decided to switch everything to UV, but it feels problematic from a higher level UI perspective. `python myscript.py` now becomes `some-tool-name run myscript.py`, so it (a) introduces a layer on top of Python (not ideal for newcomers) and (b) binds you into a specific, non-general invocation pattern.

As I mentioned before, I suspect this pattern might break a lot of existing tutorial advice, scripts with `#!/usr/bin/env python`, etc. In fact, that's a question I'm very curious about: how does `uv` intend for developers to use it if they require `uv run` for everything? (I'm speaking from complete ignorance here, of course, since I haven't studied uv at all.)

I also don't like that users now need to know about `uv` and `pip` and `uv pip`. It's often possible to plot some narrow trajectory with these tools where you hide much of the complexity (and we should have a community guide from that, perhaps via PyOpenSci, if they don't already). But then things break and you have a lot of concepts to understand before you can debug, and only uv-specific resources are helpful.

Anyway, these are just my high-level concerns; I'm sure they can all be addressed in one way for another, and I appreciate you having patience with me as I iterate them.

Reply:
Adding to what Stefan said --- and I also never used uv --- how does the workflow with a test script in a separate repository work with auto-activation?

```
$ mamba activate scipy-dev
$ python dev.py shell
$ cd ~/tmp/
$ python failing_script_with a breakpoint.py
```

Same question about working on a project (scipy) and its dependency (scipy-doctest):

```
$ mamba activate scipy-dev
$ cd scipy-doctest
$ hack hack hack
# ..... in a second terminal window
$ mamba activate scipy-dev
$ cd scipy
$ python dev.py test 
```

Reply:
> Same question about working on a project (scipy) and its dependency (scipy-doctest)

The one word answer is "workspaces" (implemented in `uv`, see https://docs.astral.sh/uv/concepts/projects/workspaces/, work in progress in `pixi`). In `pixi` right now you can do what you must also do with pip/conda: you manually install one then develop the other, without tracking the dependency between these packages. There is no good way to do it today outside of workspaces (and no, editable installs don't count since those are inherently incomplete).

Reply:
Uhm, it can't be that uncommon to have to fix a dependency? The behavior is
- in the environment I've an editable install of a dependency (scipy-doctest, but it can be numpy or anything)
- in one terminal I'm in the dependency tree, in the "bugfix" branch where I'm adding fixes
- in the other terminal I'm in the scipy tree, where I'm running scipy tests; might be also adding fixes to match

That it's an editable install is not very relevant (can be any other way to install the dependency _in this specific environment_), the choice of the env manager is irrelevant (can be mamba or venv or virtualenvwrapper or ...), the pair of projects can be anything. The main point is the ability to have two unrelated project / git trees in the same environment.

Reply:
I think Python has to become uv run everywhere.

Reply:
> but it can be numpy or anything

You actually can't with editable installs of `numpy` and say `scipy` or `scikit-learn`, it won't build the package that depends on `numpy`.

> The main point is the ability to have two unrelated project / git trees in the same environment.

That is correct. This is about the same with both methods (ignoring workspaces). In the `pip`/`conda` case you have to do `cd pkg1 && pip install .` then `cd pkg2 && pip install .  # or similar`, manually keeping track of the two packages both having a dev version installed, then running your tests. If you iteratively work on `pkg1` and `pkg2`, you just have to manage the state in your head. You can do that with `pixi` or `uv` as well - it's maybe very slightly more clumsy.

> Uhm, it can't be that uncommon to have to fix a dependency?

In the grand scheme of things, it is uncommon (>95% of users only consume rather than develop packages to start with). It's harder for project-based tools, hence I think it wasn't their first priority. But the workspaces concept was basically copied from Cargo, and seems like a good answer conceptually.